tests/hla/test_apriori.py

```diff
@@ -0,0 +1,92 @@
+                                         'ica9t3020', 'icnw34040','ID6Y05010'])
+    def test_acs_apriori(self, dataset):
+        """This test will perform fits between ALL a priori solutions and GAIA.
+
+        Success criteria:
+          * Successful fit to GAIA for dataset
+          * Fit was not compromised (fit_qual != 5)
+          * radial offset for new WCS < radial offset for IDC_* WCS
+          * scale for new WCS within 0.1% of scale for IDC_* WCS
+          * rotation for new WCS is within 0.1% of rotation from IDC_* WCS
+
+        ASSUMPTIONS:
+          * OPUS-based WCS solutions are ignored
+          * Some WCS's may be defined for one image in an ASN but not another,
+            in which case, that WCS is ignored (silently).
+
+        """
+        # Perform alignment of all WCS solutions with GAIA
+        results_dict = testutils.compare_wcs_alignment(dataset)
+        limit = 0.001
+
+        # Now compare results to see whether the a priori solutions succeeded
+        # in improving the astrometry compared to default telescope pointing
+        # reported by pipeline-defined WCS (IDC_* solution)
+        wcsnames = list(results_dict.keys())
+        idc_name = None
+        for wcs in wcsnames:
+            if 'IDC' in wcs and '-' not in wcs:
+                idc_name = wcs
+                wcsnames.remove(idc_name)
+                break
+        if idc_name is None:
+            raise ValueError
+
+        # Define pipeline-default fit
+        pipeline_results = results_dict[idc_name]
+        pipeline_offset = np.sqrt(pipeline_results['offset_x']**2 + pipeline_results['offset_y']**2)
+
+        success = False
+        # compare each WCS fit results with pipeline-default fit
+        for wcs in wcsnames:
+            print("Comparing fit for WCS='{}'...".format(wcs), end=' ')
+            results = results_dict[wcs]
+
+            # Check that fit for this WCS was successful
+            status = (results['status'] == 0).all()
+            # check that fit was not compromised or otherwise invalid
+            fit_qual =  (results['fit_qual'] < 5).all()
+
+            # Check radial offset for this WCS compared to radial offset for IDC* WCS
+            offset = np.sqrt(results['offset_x']**2+results['offset_y']**2)
+            delta = (np.abs(offset - pipeline_offset) < 1).all() and (offset > pipeline_offset).all()
+            delta = delta or (offset < pipeline_offset).all()
+
+            # Check that rotation and scale are within
+            delta_rot = np.abs(results['rotation']-pipeline_results['rotation'])/pipeline_results['rotation']
+            delta_scale = np.abs(results['scale']-pipeline_results['scale'])/pipeline_results['scale']
+            rot = (delta_rot < limit).all()
+            scale = (delta_scale < limit).all()
+
+            # Determine success/failure of this dataset's fit
+            wcs_success = status and fit_qual and delta and rot and scale
+            if wcs_success:
```

Comment:
Instead of `success = wcs_success or success` on line 90, just add
```python
success = True
```
here.

Reply:
This logic on line 90 needs to stay as originally written since we are not guaranteed that ANY of the WCS solutions will succeed.  However, we want recognize success when even just one succeeds in meeting the comparison criteria.

Reply:
Yes, and that is what the proposed change does: whenever `wcs_success` is `True` , `status` is set to `True` as well. It cannot be set to False after that. This guarantees that `success` is `True` even when only one `wcs_success` is true.

Reply:
Done...